code/game/objects/items/AI_modules/hacked.dm

```diff
@@ -34,3 +34,57 @@
 			law_datum.replace_random_law(laws[1], list(LAW_ION, LAW_HACKED, LAW_INHERENT, LAW_SUPPLIED), LAW_HACKED)
 	return laws[1]
 
+/obj/item/ai_module/malf // Gives syndie laws as well as making AI malf
+	name = "Infected AI Module"
+	desc = "An virus-infected AI Module."
+	bypass_law_amt_check = TRUE
+	laws = list("")
+	///Is this upload board unused?
+	var/functional = TRUE
+
+/obj/item/ai_module/malf/transmitInstructions(datum/ai_laws/law_datum, mob/sender, overflow)
+	if(!sender.mind?.has_antag_datum(/datum/antagonist/traitor))
+		to_chat(sender, span_warning("You have no clue how to use this thing."))
+		return
+	if(!functional)
+		to_chat(sender, span_warning("It is broken and non-functional, what do you want from it?"))
+	var/mob/living/silicon/ai/malf_candidate = law_datum.owner
+	if(!istype(malf_candidate)) //If you are using it on cyborg upload console or a cyborg
+		to_chat(sender, span_warning("You should use [src] on an AI upload console or the AI core itself."))
+		return
+	if(malf_candidate.mind?.has_antag_datum(/datum/antagonist/malf_ai)) //Already malf
+		to_chat(sender, span_warning("Unknown error occured. Upload process aborted."))
+		return
+
+	malf_candidate.laws = new /datum/ai_laws/syndicate_override
+
+	var/datum/antagonist/malf_ai/malf_datum = new (give_objectives = FALSE)
+	malf_datum.name = "Infected AI"
+	malf_datum.employer = "Infected AI"
+	malf_datum.give_zeroth_laws = FALSE
```

Comment:
Why not make a subtype of the Malf AI Datum with this information already baked in? We can even do the whole objective setup in its New or Initialize or whatever equivalent it has to "set up this datum", rather than bake it all into `transmitInstructions`.

Reply:
Done